File: src/test/java/net/slipp/domain/qna/DifferenceTagsTest.java
package net.slipp.domain.qna;

import static org.hamcrest.CoreMatchers.is;
import static org.junit.Assert.assertThat;

import java.util.Set;

import net.slipp.domain.tag.Tag;

import org.junit.Test;

import com.google.common.collect.Sets;

public class DifferenceTagsTest {
	@Test
	public void differenceTag() throws Exception {
		Set<Tag> oldTags = Sets.newHashSet(Tag.newTag("java"), Tag.newTag("javascript"));
		Set<Tag> newTags = Sets.newHashSet(Tag.newTag("java"), Tag.newTag("자바"));
		DifferenceTags tags = new DifferenceTags(oldTags, newTags);
		
		Set<Tag> taagedNewTags = Sets.newHashSet(Tag.newTag("자바"));
		Set<Tag> detaggedOldTags = Sets.newHashSet(Tag.newTag("javascript"));
		assertThat(tags.taggedNewTags(), is(taagedNewTags));
		assertThat(tags.detaggedOldTags(), is(detaggedOldTags));
	}
}

File: src/test/scala/net/slipp/domain/qna/DifferenceTagsTest.scala
package net.slipp.domain.qna

import java.util.Set

import com.google.common.collect.Sets
import net.slipp.domain.tag.Tag
import org.hamcrest.CoreMatchers._
import org.junit.Assert.assertThat
import org.junit.Test

class DifferenceTagsTest {
  @Test def differenceTag {
    val oldTags: Set[Tag] = Sets.newHashSet(Tag.newTag("java"), Tag.newTag("javascript"))
    val newTags: Set[Tag] = Sets.newHashSet(Tag.newTag("java"), Tag.newTag("자바"))
    val tags: DifferenceTags = new DifferenceTags(oldTags, newTags)
    val taagedNewTags: Set[Tag] = Sets.newHashSet(Tag.newTag("자바"))
    val detaggedOldTags: Set[Tag] = Sets.newHashSet(Tag.newTag("javascript"))
    assertThat(tags.taggedNewTags, is(taagedNewTags))
    assertThat(tags.detaggedOldTags, is(detaggedOldTags))
  }
}
